extensions/ql-vscode/src/codeql-task-provider.ts

```diff
@@ -0,0 +1,172 @@
+import * as vscode from 'vscode';
+import { CodeQLCliServer } from './cli';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+import { getOnDiskWorkspaceFolders } from './helpers';
+import { DisposableObject } from './pure/disposable-object';
+import { UserCancellationException } from './commandRunner';
+
+interface CodeQLTaskDefinition extends vscode.TaskDefinition {
+  /**
+   * The command to run.
+   */
+  command: string;
+
+  /**
+   * Additional command arguments.
+   */
+  commandArgs: string[];
+
+  /**
+   * User-specified properties for the task.
+   */
+  prompts: Prompt[];
```

Comment:
Playing around with this, using the `prompts` makes for some odd interactions if a user saves the task to the tasks.json. I think better would be to create some reasonable defaults and then allow the user to edit manually.